core/include/scipp/core/element/arithmetic.h

```diff
@@ -47,10 +60,21 @@ template <class... Ts> struct divide_types_t {
   using types = arithmetic_type_pairs;
 };
 
-constexpr auto plus =
-    overloaded{add_types_t{}, [](const auto a, const auto b) { return a + b; }};
-constexpr auto minus =
-    overloaded{add_types_t{}, [](const auto a, const auto b) { return a - b; }};
+constexpr auto plus = overloaded{
+    add_types_t{}, [](const auto a, const auto b) { return a + b; },
+    [](const scipp::core::time_point &a, const scipp::core::time_point &b) {
+      // a + b for time_point is undefined. Can't throw from here
+      // so return false
+      (void)a;
```

Comment:
Why is all this necessary? You should not have to do anything, just make sure those types are not part of `add_types_t`. If they are, something is wrong anyway, and this would be a hack to make it compile.

Reply:
`add_types_t` defines types for both `+` and `-`. I want the types to be supported for subtraction but not for addition. Don't I need to create overloads for both operations then?

Reply:
Yes, see comment https://github.com/scipp/scipp/pull/1225#pullrequestreview-453090994.